dependency.lic

```diff
@@ -772,257 +773,467 @@ class SlackbotManager
+      is_native_tm = spell_setting['skill'] == 'Targeted Magic'
+      is_sorcery_tm = spell_setting['skill'] == 'Sorcery' && spell_data['skill'] == 'Targeted Magic'
+      if is_native_tm || is_sorcery_tm
+        spell_setting['prep'] ||= 'target'
+      end
+    end
+
+    # Battle Cries
+    settings.battle_cries.map! do |battle_cry_setting|
+      battle_cry_data = battle_cries_data[battle_cry_setting['name']]
+      (battle_cry_data || {}).merge(battle_cry_setting)
+    end
+
+    # Lootables
+    settings.lootables = (
+      base_data_items.lootables +
+      base_data_items.box_nouns +
+      base_data_items.gem_nouns +
+      base_data_items.scroll_nouns +
+      settings.loot_additions -
+      settings.loot_subtractions
+    ).uniq
+
+    # Pull sensitive settings from UserVars.
+    # This is legacy from when people would commit and share their
+    # yaml files on GitHub so that specific rooms and names were not exposed.
+    settings.crossing_training_sorcery_room ||= UserVars.crossing_training_sorcery_room
+    settings.compost_room ||= UserVars.compost_room
+    settings.engineering_room ||= UserVars.engineering_room
+    settings.outfitting_room ||= UserVars.outfitting_room
+    settings.alchemy_room ||= UserVars.alchemy_room
+    settings.safe_room ||= UserVars.safe_room
+    settings.safe_room_id ||= UserVars.safe_room_id
+    settings.safe_room_empath ||= UserVars.safe_room_empath
+    settings.safe_room_empaths += (UserVars.safe_room_empaths || [])
+    settings.slack_username ||= UserVars.slack_username
+    settings.bankbot_name ||= UserVars.bankbot_name
+    settings.bankbot_room_id ||= UserVars.bankbot_room_id
+    settings.prehunt_buffs ||= UserVars.prehunt_buffs
+    settings.hometown ||= UserVars.hometown
+    settings.hometown = $HOMETOWN if $HOMETOWN
+
+    # If these room settings are defined as maps with a
+    # room id configured per town, then grab the hometown room id
+    # and assign it directly to the setting.
+    [
+      'alchemy_room',
+      'bankbot_room_id',
+      'compost_room',
+      'crossing_training_sorcery_room',
+      'enchanting_room',
+      'engineering_room',
+      'feed_cloak_room',
+      'forage_override_room',
+      'lockpick_room_id',
+      'outdoor_room',
+      'outfitting_room',
+      'prehunt_buffing_room',
+      'safe_room',
+      'safe_room_id',
+      'theurgy_prayer_mat_room'
+    ].each do |setting_name|
+      if settings[setting_name].is_a?(Hash)
+        hometown_value = settings[setting_name][settings.hometown]
+        settings[setting_name] = hometown_value if hometown_value
+      end
+    end
 
     # Merge legacy appraisal settings into target appraisal settings
-    unless s.appraisal_training.include?('pouches')
-      s.appraisal_training.unshift('pouches') if s.train_appraisal_with_pouches
+    unless settings.appraisal_training.include?('pouches')
+      settings.appraisal_training.unshift('pouches') if settings.train_appraisal_with_pouches
```

Comment:
Why are we using `unshift` over `append`?

Reply:
I did some digging on the interwebs, seems that `append` as an alias for `unshift` was introduced in [Ruby 2.5](https://rubyreferences.github.io/rubychanges/2.5.html) so it may have been that the developer at the time either (a) wasn't yet on 2.5, or (b) was accustomed to using `unshift`.

I'll update them to `append` as that alias seems more intuitive now